### ISSUE TITLE
screen_view are tracked when app is not active in firebase 6.29.0

### ISSUE THREAD

**Issue by seki-ne** [firebase/firebase-ios-sdk#6327]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.4.1
  * Firebase SDK version: 6.29.0, 6.30.0
  * Firebase Component: Analytics
  * Component version: 6.7.0
  * Installation method: `CocoaPods`

### [REQUIRED] Step 2: Describe the problem
i use background fetch.
`screen_view` are tracked when app is not active in firebase 6.29.0, 6.30.0
i also test  in 6.24.0 and `screen_view` are not tracked.
```
2020-08-24 14:01:26.551486+0900 MyAPP[57082:16580280] 6.24.0 - [Firebase/Analytics][I-ACS031013] Screen view event not logged. App is not active.
```

i expect events not to be logged if app is not active.

#### Steps to reproduce:
edit scheme -> options -> check `launch due to a background fetch event`
```
AppDelegate.swift
func application(_: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
// do something
}
```
```
class ViewController: UIViewController {
  override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        Analytics.logEvent(AnalyticsEventScreenView,
                   parameters: [AnalyticsParameterScreenName: screenName,
                                AnalyticsParameterScreenClass: screenClass])
  }
}
```
screen_views will be tracked when app is not active.(v6.29.0, v6.30.0)

Related issues: https://github.com/firebase/firebase-ios-sdk/issues/1894

**Comment by seki-ne** [comment]
i make a sample to reproduce it
https://github.com/seki-ne/myapp

**Comment by paulb777** [comment]
Thanks for the repro! Filed internally at b/166440923

**Comment by allenktv** [comment]
@seki-ne Thanks for the report!

In version 6.24.0, the manual `screen_view` was not implemented yet, so it makes sense that it doesn't get logged, however I saw that the automatic `screen_view` was logged back in the old version, as well as the new one, so there is no regression there. 

In terms of why the `screen_view` event was being called in the first place: I saw that the OS calls `viewDidAppear` on the ViewController, even though the view hasn't actually appeared. This to me seems like a bug with how Apple addresses the `launch due to a background fetch event` option and how ViewController lifecycle is handled. Because of this, I am hesitant to call this an issue on our end unless it is reproducible in an actual app without that option checked, so I will close this issue for now.
